src/inc/msquic.hpp

```diff
@@ -121,14 +224,35 @@ class MsQuicRegistration {
     bool IsValid() const { return Registration != nullptr; }
     MsQuicRegistration(MsQuicRegistration& other) = delete;
     MsQuicRegistration operator=(MsQuicRegistration& Other) = delete;
-    operator HQUIC () {
+    operator HQUIC () const {
         return Registration;
     }
 };
 
 struct MsQuicSession {
     HQUIC Handle;
     bool CloseAllConnectionsOnDelete;
+
+    MsQuicSession(
+        _In_ const MsQuicRegistration& Reg,
+        _In_z_ const char* RawAlpn = "MsQuicTest")
+        : Handle(nullptr), CloseAllConnectionsOnDelete(false) {
+        QUIC_BUFFER Alpn;
+        Alpn.Buffer = (uint8_t*)RawAlpn;
+        Alpn.Length = (uint32_t)strlen(RawAlpn);
+        if (QUIC_FAILED(
+            MsQuic->SessionOpen(
+                Reg,
+                &Alpn,
+                1,
+                nullptr,
+                &Handle))) {
+            Handle = nullptr;
+        }
+    }
+
+#ifndef QUIC_SKIP_GLOBAL_CONSTRUCTORS
```

Comment:
Why?

Reply:
Because otherwise I have to define an extern Registration somewhere, because these are using globals. And fixing that requires massively changing the BVTs.

Reply:
Ok.